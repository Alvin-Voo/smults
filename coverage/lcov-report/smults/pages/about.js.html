<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for smults/pages/about.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">smults/pages</a> about.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>1/1</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';// work with components
import Layout from '../components/Layout';
&nbsp;
export default () =&gt; (
  &lt;Layout item="about"&gt;
    &lt;div className="aboutDiv"&gt;
      &lt;h2&gt;Intro video&lt;/h2&gt;
      &lt;iframe
        title="youtube-intro"
        width="750"
        height="425"
        src="https://www.youtube.com/embed/SWYt4bGmN7w"
        frameBorder="0"
        allow="autoplay; encrypted-media"
        allowFullScreen
      /&gt;
    &lt;/div&gt;
    &lt;div className="aboutDiv"&gt;
      &lt;h3&gt;Why smults?&lt;/h3&gt;
      &lt;p&gt;
        &lt;a href="https://steemit.com" target="_blank" rel="noopener noreferrer"&gt;steemit.com&lt;/a&gt;
        {' '}
        allows searches by using only ONE tag, while
        {' '}
        &lt;a href="https://busy.org/" target="_blank" rel="noopener noreferrer"&gt;busy.org&lt;/a&gt;
        {' '}
        searches by single user&amp;apos;s name,
        this is inconvenient for those who need to look for more specific posts.
        Smults is here to address this issue!
      &lt;/p&gt;
      &lt;h3&gt;Can I add custom tags in the search bar?&lt;/h3&gt;
      &lt;p&gt;Sure! Just start typing and press &amp;apos;Enter&amp;apos; to include the tag(s) when you are done!&lt;/p&gt;
      &lt;h3&gt;Why &amp;apos;15&amp;apos; trending tags to start with? Why not &amp;apos;18&amp;apos; or &amp;apos;23&amp;apos;?&lt;/h3&gt;
      &lt;p&gt;Well.. That&amp;apos;s just my personal preference ;). You can add more custom search tags by youself!&lt;/p&gt;
      &lt;h3&gt;What is this &amp;apos;Mark first tag as category&amp;apos; checkbox thing?&lt;/h3&gt;
      &lt;p&gt;When you mark it and search, the return results will have their categories as the first tag you are searching for. Isn&amp;apos;t that just convenient? ;)&lt;/p&gt;
      &lt;h3&gt;I spotted a nasty error/bug.. Where can I report it?&lt;/h3&gt;
      &lt;p&gt;
        Head over to this project&amp;apos;s
        &lt;a href="https://github.com/Alvin-Voo/smults" target="_blank" rel="noopener noreferrer"&gt;github page&lt;/a&gt;
        {' '}
        and log an issue.
      &lt;/p&gt;
      &lt;h3&gt;I have a great idea for your application! I want to contribute!&lt;/h3&gt;
      &lt;p&gt;
        Great! Let me know at
        &lt;a href="https://steemit.com/@alvinvoo" target="_blank" rel="noopener noreferrer"&gt;@alvinvoo&lt;/a&gt;
        , comment on my post which announces this project, or you can visit this project&amp;apos;s
        &lt;a href="https://github.com/Alvin-Voo/smults" target="_blank" rel="noopener noreferrer"&gt;github page&lt;/a&gt;
        {' '}
        and log an issue but with the title starting with &amp;apos;[FEATURE]&amp;apos;
      &lt;/p&gt;
      &lt;h3&gt;Is this free?&lt;/h3&gt;
      &lt;p&gt;Of course! I will let you know when it&amp;apos;s not. ;)&lt;/p&gt;
    &lt;/div&gt;
  &lt;/Layout&gt;
);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Sep 16 2018 11:24:24 GMT+0800 (+08)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
