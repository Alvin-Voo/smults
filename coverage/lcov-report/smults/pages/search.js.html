<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for smults/pages/search.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">smults/pages</a> search.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>26/26</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>12/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>25/25</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140
141
142
143
144
145
146
147
148
149
150
151
152
153
154
155
156</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">15x</span>
<span class="cline-any cline-yes">15x</span>
<span class="cline-any cline-yes">15x</span>
<span class="cline-any cline-yes">15x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">45x</span>
<span class="cline-any cline-yes">45x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">45x</span>
<span class="cline-any cline-yes">45x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';// work with components
import PropTypes from 'prop-types';
import {
  Dropdown, Button, Checkbox, Popup, Icon,
} from 'semantic-ui-react';
import { take } from 'lodash';
import { connect } from 'react-redux';
import Layout from '../components/Layout';
import ConnectedPostList from '../components/PostList';
import ScrollButton from '../components/ScrollButton';
import { fetchTrendingTags, fetchPosts } from '../actions';
&nbsp;
export class Search extends Component {
  state = {
    tagsOptions: [],
    selectedTags: [],
    filterOptions: [
      { text: 'New', value: 'created' },
      { text: 'Trending', value: 'trending' },
      { text: 'Hot', value: 'hot' },
      { text: 'Active', value: 'active' },
      { text: 'Promoted', value: 'promoted' },
    ],
    selectedFilter: 'created',
    checkedCategory: false,
    searchLoading: false,
    error: '',
  }
&nbsp;
  componentDidMount() {
    const { fetchTrendingTags } = this.props;
    fetchTrendingTags().then(() =&gt; {
      const { tagsOptions } = this.props;
      this.setState({ tagsOptions });
    });// we only want to fetch in client side once
  }
&nbsp;
  dropDownChange = (e, d) =&gt; {
    this.setState({ selectedTags: take(d.value, 5) });
  }
&nbsp;
  dropDownAddItem = (e, d) =&gt; {
    this.setState((prevState) =&gt; {
      const newTags = prevState.tagsOptions;
      newTags.unshift({ key: d.value, value: d.value, text: d.value });
      return { tagsOptions: newTags };
    });
  }
&nbsp;
  search = () =&gt; {
    const { selectedTags, selectedFilter, checkedCategory } = this.state;
    const { fetchPosts } = this.props;
    if (selectedTags.length === 0) return;
    this.setState({ searchLoading: true, error: '' });
    fetchPosts(selectedTags, selectedFilter, checkedCategory).catch(
      (error) =&gt; {
        this.setState({ error: String(error) });
      },
    );
    this.setState({ searchLoading: false });
  }
&nbsp;
  filterChange = (e, d) =&gt; {
    this.setState({ selectedFilter: d.value });
  }
&nbsp;
  checkBoxCategory = (e, d) =&gt; {
    this.setState({ checkedCategory: d.checked });
  }
&nbsp;
  displayError = () =&gt; {
    const { error } = this.state;
    if (error) {
      return (
        &lt;div className="error"&gt;
          &lt;p&gt;{error}&lt;/p&gt;
          &lt;p&gt;
            Sorry, something is broken. Take a screenshot and notify
            &lt;a href="https://steemit.com/@alvinvoo" target="_blank" rel="noopener noreferrer"&gt;
              {' '}
              @alvinvoo
            &lt;/a&gt;
          &lt;/p&gt;
        &lt;/div&gt;
      );
    }
  }
&nbsp;
  render() {
    const {
      tagsOptions, selectedTags, filterOptions, selectedFilter, searchLoading, checkedCategory,
    } = this.state;
    return (
      &lt;Layout item="search"&gt;
        &lt;div className="searchBar"&gt;
          &lt;Dropdown
            icon="search"
            placeholder="Insert your tags here..."
            selection
            multiple
            search
            fluid
            allowAdditions
            options={tagsOptions}
            onAddItem={this.dropDownAddItem}
            onChange={this.dropDownChange}
            value={selectedTags}
          /&gt;
          &lt;Button
            positive
            loading={searchLoading}
            onClick={this.search}
          &gt;
            {' '}
            Search
          &lt;/Button&gt;
        &lt;/div&gt;
        &lt;div className="searchOptionsBar"&gt;
          &lt;div&gt;
          Filter tags by
            {' '}
            {' '}
            &lt;Dropdown
              inline
              options={filterOptions}
              onChange={this.filterChange}
              value={selectedFilter}
            /&gt;
          &lt;/div&gt;
          &lt;div className="markyTag"&gt;
            &lt;Checkbox label="Mark first tag as category" onChange={this.checkBoxCategory} checked={checkedCategory} /&gt;
            &lt;Popup trigger={&lt;Icon name="question circle outline" /&gt;} content="Return search results which match the given first tag as their categories." /&gt;
          &lt;/div&gt;
        &lt;/div&gt;
        &lt;div className="postList"&gt;
          &lt;ConnectedPostList /&gt;
          {this.displayError()}
        &lt;/div&gt;
        &lt;ScrollButton scrollStepInPx={50} delayInMs={16.66} /&gt;
      &lt;/Layout&gt;
    );
  }
}
&nbsp;
Search.propTypes = {
  fetchPosts: PropTypes.func.isRequired,
  fetchTrendingTags: PropTypes.func.isRequired,
  tagsOptions: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.string)).isRequired,
};
&nbsp;
export function mapStateToProps({ tags }) {
  return { tagsOptions: tags.trending_tags_options };
}
&nbsp;
export default connect(mapStateToProps, { fetchTrendingTags, fetchPosts })(Search);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Sep 16 2018 11:24:24 GMT+0800 (+08)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
