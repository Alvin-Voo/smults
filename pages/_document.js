// ./pages/_document.js
import Document, { Head, Main, NextScript } from 'next/document';

export default class MyDocument extends Document {
  render() {
    return (
      <html>
        <Head>
          <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.3.1/semantic.min.css" />
          <link rel="stylesheet" href="/_next/static/style.css" />
          <link rel="shortcut icon" href="/static/favicon.ico" type="image/x-icon"></link>
        </Head>
        <body>
          <Main />
          <NextScript />
        </body>
      </html>
    );
  }
}
