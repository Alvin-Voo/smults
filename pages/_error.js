import React from 'react';
import PropTypes from 'prop-types';
import Error from 'next/error';

export default class MyError extends React.Component {
  static async getInitialProps({ req, res }) {
    const { statusCode } = res;

    return { statusCode };
  }

  render() {
    const { statusCode } = this.props;
    return (
      <div className="errorPage">
        <a href="/">Return to Home Page</a>
        <Error statusCode={statusCode} />
      </div>
    );
  }
}

MyError.propTypes = {
  statusCode: PropTypes.number.isRequired,
}
