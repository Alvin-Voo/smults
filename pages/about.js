import React, {Component} from 'react';//work with components
import Layout from '../components/Layout';

class About extends Component{

  render(){
    return(
      <Layout item='about'>
        About
      </Layout>
    )
  }

}

export default About;
