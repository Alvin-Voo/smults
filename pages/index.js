import React, { Component } from 'react';// work with components
import { connect } from 'react-redux';
import ConnectedSearch from './search';


export class Index extends Component {
  static getInitialProps({ reduxStore, req }) {
    // this is the initial props function in server
    const isServer = !!req;
    return { isServer };
  }

  render() {
    return <ConnectedSearch />;
  }
}

export default connect()(Index);
