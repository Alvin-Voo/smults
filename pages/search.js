import React, { Component } from 'react';// work with components
import PropTypes from 'prop-types';
import {
  Dropdown, Button, Checkbox, Popup, Icon,
} from 'semantic-ui-react';
import { take } from 'lodash';
import { connect } from 'react-redux';
import Layout from '../components/Layout';
import ConnectedPostList from '../components/PostList';
import ScrollButton from '../components/ScrollButton';
import { fetchTrendingTags, fetchPosts } from '../actions';

export class Search extends Component {
  state = {
    tagsOptions: [],
    selectedTags: [],
    filterOptions: [
      { text: 'New', value: 'created' },
      { text: 'Trending', value: 'trending' },
      { text: 'Hot', value: 'hot' },
      { text: 'Active', value: 'active' },
      { text: 'Promoted', value: 'promoted' },
    ],
    selectedFilter: 'created',
    checkedCategory: false,
    searchLoading: false,
    error: '',
  }

  componentDidMount = async () => {
    const { fetchTrendingTags } = this.props;
    await fetchTrendingTags();
    // we only want to fetch in client side once
    const { tagsOptions } = this.props;
    this.setState({ tagsOptions });
  }

  dropDownChange = (e, d) => {
    this.setState({ selectedTags: take(d.value, 5) });
  }

  dropDownAddItem = (e, d) => {
    this.setState((prevState) => {
      const newTags = prevState.tagsOptions;
      newTags.unshift({ key: d.value, value: d.value, text: d.value });
      return { tagsOptions: newTags };
    });
  }

  search = async () => {
    const { selectedTags, selectedFilter, checkedCategory } = this.state;
    const { fetchPosts } = this.props;
    if (selectedTags.length === 0) return;
    this.setState({ searchLoading: true, error: '' });
    await fetchPosts(selectedTags, selectedFilter, checkedCategory).catch(
      (error) => {
        this.setState({ error: String(error) });
      },
    );
    this.setState({ searchLoading: false });
  }

  filterChange = (e, d) => {
    this.setState({ selectedFilter: d.value });
  }

  checkBoxCategory = (e, d) => {
    this.setState({ checkedCategory: d.checked });
  }

  displayError = () => {
    const { error } = this.state;
    if (error) {
      return (
        <div className="error">
          <p>{error}</p>
          <p>
            Sorry, something is broken. Take a screenshot and notify
            <a href="https://steemit.com/@alvinvoo" target="_blank" rel="noopener noreferrer">
              {' '}
              @alvinvoo
            </a>
          </p>
        </div>
      );
    }
  }

  render() {
    const {
      tagsOptions, selectedTags, filterOptions, selectedFilter, searchLoading, checkedCategory,
    } = this.state;
    return (
      <Layout item="search">
        <div className="searchBar">
          <Dropdown
            icon="search"
            placeholder="Insert your tags here..."
            selection
            multiple
            search
            fluid
            allowAdditions
            options={tagsOptions}
            onAddItem={this.dropDownAddItem}
            onChange={this.dropDownChange}
            value={selectedTags}
          />
          <Button
            positive
            loading={searchLoading}
            onClick={this.search}
          >
            {' '}
            Search
          </Button>
        </div>
        <div className="searchOptionsBar">
          <div>
          Filter tags by
            {' '}
            {' '}
            <Dropdown
              inline
              options={filterOptions}
              onChange={this.filterChange}
              value={selectedFilter}
            />
          </div>
          <div className="markyTag">
            <Checkbox label="Mark first tag as category" onChange={this.checkBoxCategory} checked={checkedCategory} />
            <Popup trigger={<Icon name="question circle outline" />} content="Return search results which match the given first tag as their categories." />
          </div>
        </div>
        <div className="postList">
          <ConnectedPostList />
          {this.displayError()}
        </div>
        <ScrollButton scrollStepInPx={50} delayInMs={16.66} />
      </Layout>
    );
  }
}

Search.propTypes = {
  fetchPosts: PropTypes.func.isRequired,
  fetchTrendingTags: PropTypes.func.isRequired,
  tagsOptions: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.string)).isRequired,
};

export function mapStateToProps({ tags }) {
  return { tagsOptions: tags.trending_tags_options };
}

export default connect(mapStateToProps, { fetchTrendingTags, fetchPosts })(Search);
