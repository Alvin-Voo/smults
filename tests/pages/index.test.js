import React from 'react';
import { shallow } from 'enzyme';
import { Index } from '../../pages/index';

test('should render Index page properly ', () => {
  const wrapper = shallow(<Index />);
  expect(wrapper).toMatchSnapshot();
});

test('should have a static getInitialProps method for connected Index', async () => {
  const ret = await Index.getInitialProps({ reduxStore: '', req: true });
  expect(ret).toEqual({ isServer: true });
});
