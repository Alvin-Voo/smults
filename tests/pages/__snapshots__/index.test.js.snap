// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Index page properly  1`] = `<Connect(Search) />`;
